containers/src/Kokkos_DynRankView.hpp

```diff
@@ -128,7 +129,20 @@ struct DynRankDimTraits {
         layout.dimension[5] != unspecified ? layout.dimension[5] : 1,
         layout.dimension[6] != unspecified ? layout.dimension[6] : 1,
         layout.dimension[7] != unspecified ? layout.dimension[7] : unspecified);
-    new_layout.stride = layout.stride;
+    // In order to have a valid LayoutRight stride when Sacado passes through
+    // extra integer arguments, we need to set it to the dimension[6] for the
+    // rank-7 view coming out of here. Only if the original layout was already
+    // rank-7 we can preserve the stride.
```

Comment:
uggg ...

Reply:
Why is `Sacado` ever using the generic template instead of its specialization for `Kokkos::Impl::ViewSpecializeSacadoFad`? 

Does the change here unify the specialization with the generic template so that it doesn't need to be provided in `Trilinos` anymore?

Reply:
That file is defect unused in my new implementation. And the path to getting this out is that users HAVE to specify the fad dimension via an argument passed to `view_alloc` instead of an extra int argument in random places.